db/substate_encoding.go

```diff
@@ -69,6 +86,14 @@ func (db *substateDB) decodeToSubstate(bytes []byte, block uint64, tx int) (*sub
 	return db.encoding.decode(bytes, block, tx)
 }
 
+// encodeSubstate defensively defaults to "default" if nil
+func (db *substateDB) encodeSubstate(ss *substate.Substate, block uint64, tx int) ([]byte, error) {
+	if db.encoding == nil {
+		db.SetSubstateEncoding("default")
+	}
+	return db.encoding.encode(ss, block, tx)
+}
```

Comment:
First I think implementation details does not belong to a func comment.
Second, do you think we should have this condition? I think this belongs into the factory func.

Reply:
Defensive defaults now removed in favor of having `SetSubstate("default")` in all public constructor of `substateDB`.
[79974a4](https://github.com/Fantom-foundation/Substate/pull/85/commits/79974a431fa86c7484b5996c1cf472b923e4bd2a)

I understand the performance reason to do this, but this leaves direct instantiation of `&substateDB{...}` to panic if encoding is never set. Any advice on this?